data/adapters/helpers.py

```diff
@@ -128,3 +128,151 @@ def to_float(str):
+
+
+def get_publication_ids(portal_url, object):
+    publication_ids = set()
+    publications = object.get('publications')
+    for publication in publications:
+        publication_object = requests.get(
+            portal_url + publication + '/@@object?format=json').json()
+        publication_ids.add(publication_object.get('publication_identifiers', [])[
+                            0])  # only one ID from a publication is needed
+    return publication_ids
+
+
+def query_fileset_files_props_igvf(file_accession):
+    portal_url = 'https://api.data.igvf.org/'
+    # get file metadata
+    file_object = requests.get(
+        portal_url + file_accession + '/@@object?format=json').json()
+    lab = file_object.get('lab')
+    software = set()
+    if 'analysis_step_version' in file_object:
+        analysis_step_version_object = requests.get(
+            portal_url + file_object.get('analysis_step_version') + '/@@object?format=json').json()
+        software_versions = analysis_step_version_object.get(
+            'software_versions')
+        for software_version in software_versions:
+            software_version_object = requests.get(
+                portal_url + software_version + '/@@object?format=json').json()
+            software_object = requests.get(
+                portal_url + software_version_object.get('software') + '/@@object?format=json').json()
+            software.add(software_object.get('name'))
+
+    file_set_object = requests.get(
+        portal_url + file_object.get('file_set') + '/@@object?format=json').json()
+    file_set_accession = file_set_object.get('accession')
+    file_set_object_type = file_set_object.get('@type')[0]
+
+    preferred_assay_titles = set()
+    assay_term_ids = set()
+    publication_ids = set()
+
+    # get file set metadata
+    prediction = False
+    prediction_method = None
+    if file_set_object_type == 'PredictionSet':
+        prediction = True
+        prediction_method = file_set_object.get('file_set_type')
+        if not(software):
+            raise(ValueError(f'Prediction sets require software to be loaded.'))
+    elif file_set_object_type == 'AnalysisSet':
+        for input_file_set in file_set_object.get('input_file_sets', []):
+            if input_file_set.startswith('/measurement-sets/'):
+                input_file_set_object = requests.get(
+                    portal_url + input_file_set + '/@@object?format=json').json()
+                preferred_assay_titles.add(
+                    input_file_set_object.get('preferred_assay_title'))
+                assay_term = input_file_set_object.get('assay_term')
+                assay_term_object = requests.get(
+                    portal_url + assay_term + '/@@object?format=json').json()
+                assay_term_ids.add(assay_term_object.get('term_id'))
+    else:
+        raise(ValueError(
+            f'Loading data from file sets other than prediction sets and analysis sets is currently unsupported.'))
```

Comment:
For SEMpl (Boyle lab) we will also load their model files (as motifs) under model sets, but we can add the support later.

Reply:
left a note to add support later.